server/src/main/java/org/candlepin/pinsetter/tasks/HypervisorUpdateJob.java

```diff
@@ -77,22 +80,36 @@
     private static Logger log = LoggerFactory.getLogger(HypervisorUpdateJob.class);
     private OwnerCurator ownerCurator;
     private ConsumerCurator consumerCurator;
+    private ConsumerTypeCurator consumerTypeCurator;
     private ConsumerResource consumerResource;
     private I18n i18n;
+    private SubscriptionServiceAdapter subAdapter;
+    private ComplianceRules complianceRules;
 
     public static final String CREATE = "create";
     public static final String REPORTER_ID = "reporter_id";
     public static final String DATA = "data";
     public static final String PRINCIPAL = "principal";
     protected static String prefix = "hypervisor_update_";
 
+    private ConsumerType hypervisorType;
+
+
     @Inject
     public HypervisorUpdateJob(OwnerCurator ownerCurator, ConsumerCurator consumerCurator,
-        ConsumerResource consumerResource, I18n i18n) {
+                               ConsumerTypeCurator consumerTypeCurator, ConsumerResource consumerResource, I18n i18n,
+                               SubscriptionServiceAdapter subAdapter,
+                               ComplianceRules complianceRules) {
+
         this.ownerCurator = ownerCurator;
         this.consumerCurator = consumerCurator;
+        this.consumerTypeCurator = consumerTypeCurator;
         this.consumerResource = consumerResource;
         this.i18n = i18n;
+        this.subAdapter = subAdapter;
+        this.complianceRules = complianceRules;
+
+        this.hypervisorType = consumerTypeCurator.lookupByLabel(ConsumerTypeEnum.HYPERVISOR.getLabel());
```

Comment:
We need to be a bit careful here in actual production code: we're broken if this returns null.

Reply:
Still need a contingency plan here.